src/strategy/gradual-rollout-user-id.ts

```diff
@@ -0,0 +1,23 @@
+import { Strategy } from '../strategy';
+import { Context } from '../context';
+import { normalizedValue } from './util';
+
+export class GradualRolloutUserIdStrategy extends Strategy {
```

Comment:
yes i thought about it, but i feel the abstraction costs more than the duplication. it's easier to read the code when it is explicit!